Set the GUI-selected environment in the ksdata (#1192100)

dashea opened the pull request:
This way an environment selected in the GUI and an environment selected
in kickstart are treated in the same way. This fixes the problem where
changing payloads forgets the previous environment selection, and it
changes the generated %packages section of anaconda-ks.cfg to contain
the environment information instead of a list of the groups in the
environment.

This commit restores the dnf _select_environment method removed in
602f47dcd5d7543f66dc6833685d0e810595acb5, but modifies it so that the
environment is both selected in dnf and in the kickstart data.

vpodzime commented on pyanaconda/packaging/__init__.py:
```diff
@@ -290,7 +290,11 @@ def selectGroup(self, groupid, default=True, optional=False):
+        # the group list
+        if self.data.packages.environment and \
+                groupid not in self.environmentGroups(self.data.packages.environment):
+            self.data.packages.groupList.append(grp)
```
Is it okay that we don't add the group to the list if no environment is selected? I'd expect it to be added so please extend the comment a bit to explain why that's not happening.

dashea replied:
This is ok because with this patch the environment gets added to the ksdata instead of just its groups, and the selected environment gets added to the payload via dnf.Base.environment_install or yum.selectEnvironment. I'll change the comment to "If the group is part of the environment, it is already selected as part of the environment. Do not duplicate it in the group list"

vpodzime commented on pyanaconda/packaging/yumpayload.py:
```diff
@@ -1214,7 +1231,7 @@ def _applyYumSelections(self):
         if env:
             try:
-                self.selectEnvironment(env)
+                self._selectYumEnvironment(env)
```
Is it okay that we no longer call the parent class' ``selectEnvironment()``?

dashea replied:
Probably not. I'll change that to match dnfpayload: leave the selectEnvironment call alone, and override selectEnvironment to call the parent method and then _selectYumEnvironment.

vpodzime commented:
Other than the two comments above this looks good to me.

dashea commented:
Unacking because this messes up the dnf group exclusion logic and I want to make sure I don't get distracted and forget to fix that.

dashea commented:
Redid the patch so that environment selections are applied to dnf as a series of group selections (so group exclusion works), changed PackagePayload.selectEnvrionment to just set the environment in ksdata and not do anything with groups, and changed the dnf and yum apply_selections to only call their internal environment methods instead of going back to selectEnvironment again.

vpodzime commented:
Looks good and makes sense to me.
